engine/exchange.go

```diff
@@ -375,17 +384,54 @@ func (bot *Engine) SetupExchanges() {
 			log.Debugf(log.ExchangeSys, "%s: Exchange support: Disabled\n", configs[x].Name)
 			continue
 		}
-		err := bot.LoadExchange(configs[x].Name, true, &wg)
-		if err != nil {
-			log.Errorf(log.ExchangeSys, "LoadExchange %s failed: %s\n", configs[x].Name, err)
-			continue
-		}
-		log.Debugf(log.ExchangeSys,
-			"%s: Exchange support: Enabled (Authenticated API support: %s - Verbose mode: %s).\n",
-			configs[x].Name,
-			common.IsEnabled(configs[x].API.AuthenticatedSupport),
-			common.IsEnabled(configs[x].Verbose),
-		)
+		wg.Add(1)
+		go func(iterator int) {
+			err := bot.LoadExchange(configs[iterator].Name, true, &wg)
```

Comment:
would make a config struct copy `cpy := configs[iterator]` and pass by value in the `go func( cfg confStruct) {....} (cpy)` because when your accessing configs via your iterator in different routines it becomes a race issue due to the reads.